jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -563,7 +564,7 @@ protected void handleException(Throwable failure)
                 {
                     _response.setHeader(HttpHeader.CONNECTION.asString(), HttpHeaderValue.CLOSE.asString());
 
-                    Throwable cause = unwrap(failure);
+                    Throwable cause = unwrap(failure,BadMessageException.class,UnavailableException.class);
```

Comment:
Is this an order of precedence?

What happens if you have ...

ServletException -> BadMessageException -> UnavailableException  (seems strange now that I write it out)
or
UnavailableException -> BadMessageException -> ServletException (this seems strange too)

In other words, what's important?  The existence of the nested cause? or the order of the nested cause?

Reply:
Good question.  I did think of this, but in this case it is OK because we are looking to determine the response code and it is the outermost exception that gets to decide what that is.  eg a UAE wrapping a BME, will send a 503 unavailable regardless of what the BME says.